src/convertToObject.js

```diff
@@ -12,7 +12,16 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+  const trimed = sourceString.split(';').map(element => element.trim());
+  const noEmpty = trimed.filter(word => word !== '');
+  const pairs = noEmpty.map(element => element.split(':'));
```

Comment:
You could do this using one variable and chained methods.
```suggestion
  const pairs = sourceString
    .split(';')
    .map(element => element.trim());
    .filter(word => word !== '');
    .map(element => element.split(':'));
```